domain/upgrade/service/service.go

```diff
@@ -116,13 +118,17 @@ func (s *Service) ActiveUpgrade(ctx context.Context) (domainupgrade.UUID, error)
 	return activeUpgrade, domain.CoerceError(err)
 }
 
-// WatchForUpgradeReady creates a watcher which notifies when all controller
-// nodes have been registered, meaning the upgrade is ready to start
-func (s *Service) WatchForUpgradeReady(ctx context.Context, upgradeUUID domainupgrade.UUID) (watcher.NotifyWatcher, error) {
+// WatchForUpgradeStarted creates a watcher which notifies when all controller
+// nodes have been registered, meaning the upgrade is ready to start.
+func (s *Service) WatchForUpgradeStarted(ctx context.Context, upgradeUUID domainupgrade.UUID) (watcher.NotifyWatcher, error) {
 	if err := upgradeUUID.Validate(); err != nil {
 		return nil, errors.Trace(err)
 	}
-	return NewUpgradeReadyWatcher(ctx, s.st, s.watcherFactory, upgradeUUID)
+
+	mask := changestream.Create | changestream.Update
+	return s.watcherFactory.NewValuePredicateWatcher("upgrade_info_controller_node", upgradeUUID.String(), mask, func(ctx context.Context, db coredatabase.TxnRunner, changes []changestream.ChangeEvent) (bool, error) {
```

Comment:
Could declare the predicate func separately or assign it to a variable to shorten this line.